optuna/notebooks/jupytercon_demo/RAPIDS_xfeat_Optuna-GPU.ipynb

```diff
@@ -0,0 +1,1803 @@
+    "\n",
+    "Set the `download_data=True` and `data_dir` to a local path in your system to download the data into the path of `data_dir`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "download_data = True\n",
+    "\n",
+    "file_name = 'airline_small.parquet' # NOTE: Change to airline_20000000.parquet to use a larger dataset\n",
+    "\n",
+    "data_dir = \"/home/data/airline_data/\" # NOTE: Change to a local path where you want to save the file\n",
+    "INPUT_FILE = os.path.join(data_dir, file_name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      " > File already exists. Ready to load at /home/data/airline_data/airline_small.parquet\n",
+      "Completed!\n"
+     ]
+    }
+   ],
+   "source": [
+    "if download_data:\n",
+    "    from urllib.request import urlretrieve\n",
+    "\n",
+    "    if os.path.isfile(INPUT_FILE):\n",
+    "            print(f\" > File already exists. Ready to load at {INPUT_FILE}\")\n",
+    "    else:\n",
+    "        # Ensure folder exists\n",
+    "        os.makedirs(data_dir, exist_ok=True)\n",
+    "\n",
+    "    url = \"https://rapidsai-cloud-ml-sample-data.s3-us-west-2.amazonaws.com/\" + file_name\n",
+    "\n",
+    "    urlretrieve(url= url,filename=INPUT_FILE)\n",
+    "\n",
+    "    print(\"Completed!\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### MLflow Configuration\n",
+    "\n",
+    "For tracking the hyperparameter optimisation expriments, we will use MLFlow. In the next cell, the required variables are set up along with the callback class `RAPIDSMLflowCallback` that we will pass to Optuna for tracking."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "MLFLOW_TRACKING_URI='sqlite:////tmp/mlflow-db.sqlite'\n",
+    "MLFLOW_MODEL_ID = \"rapids-optuna-airline\"\n",
+    "\n",
+    "def get_latest_mlflow_model(tracking_uri, model_id):\n",
+    "    client = MlflowClient(tracking_uri=tracking_uri, registry_uri=tracking_uri)\n",
+    "    model = client.get_registered_model(model_id)\n",
+    "    latest_model = model.latest_versions[0]\n",
+    "\n",
+    "    return f\"MLFLOW_TRACKING_URI={tracking_uri} mlflow models serve --no-conda -m models:/{model_id}/{latest_model.version} -p 56767\"\n",
+    "\n",
+    "## Custom callback, for additional flexibility, based on MLflowCallback\n",
+    "class RAPIDSMLflowCallback(object):\n",
```

Comment:
This would be nice to document especially in slides - how does this differ from the standard MLFlow callback?

Reply:
@drobison00 is getting the slides ready for this, and he will be talking about it during the walkthrough too.